### PR TITLE
 add column adjusting to result table

### PR DESCRIPTION
**Pull request by Isuru-W**
**What does this PR do, and why?**

Currently Table View on Results pane have fixed column width. Added functionality to drag column border to adjust its width. Double clicking on column border will reset its width.

**How should this be tested?**

1. Hover over a column border and drag the column to change its width.
2. Double click on the adjusted column border to reset its width.

**PR Definition of Done**

- Table view columns adjustable in Results pane without braking any current functionalities
- Minimum 2 reviewer approval 
- All comments resolved
- Hero build with results
- Passing CI or full local build

**Relevant links:**
[ISR-10834](https://jira.nciteglobal.com.au/jira/browse/ISR-10834)

**Demo:**

https://user-images.githubusercontent.com/125233437/224907156-2108df94-8ae5-4e0e-b7b7-3d10a85571ac.mp4

### CODE REVIEW

**Comment by lifenouveau** [comment]
I think the click and drag looks good visually. Does the drag point save and "remember" it's spacing for the user? Or would the user have to drag each time due to a component reset/reload?

**Comment by Isuru-W** [comment]
> I think the click and drag looks good visually. Does the drag point save and "remember" it's spacing for the user? Or would the user have to drag each time due to a component reset/reload?

**Comment by Isuru-W** [comment]
> I think the click and drag looks good visually. Does the drag point save and "remember" it's spacing for the user? Or would the user have to drag each time due to a component reset/reload?

Currently the column widths are not being saved, they would have to drag each time.

**Comment by alexabird** [comment]
build now

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/results-visual/result-item.tsx
Comment (alexabird) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/results-visual/result-item.tsx:1]: ✏️ remove all `style` references or uncomment

**Comment by alexabird** [comment]
✏️ Hero was successful
The only thing that seems a little weird is the hover/highlight not covering the whole title.
Pasted Graphic copy

**Comment by Isuru-W** [comment]
Hi @alexabird, I added a div so the column can be dragged. It is not highlighting because its not part of the button component. Does the styling need to be changed?

**Comment by alexabird** [comment]
> Hi @alexabird, I added a div so the column can be dragged. It is not highlighting because its not part of the button component. Does the styling need to be changed?

Im not sure 🤷‍♀️ I personally think the different hover cursors are enough to tell the difference between the button and the drag/resize div. Good question for @lifenouveau

**Comment by lifenouveau** [comment]
I think the current interaction is helpful as it is.

**Comment by andrewkfiedler** [comment]
Might be worth adding a debounce to the callback for mouse move handler, as even on my newer mac it struggles when there are a lot of columns.  

Other than that it looks fine.  I would prefer we put the dragger directly in the middle between the columns rather than on one side, but this works for now and we can improve on it later.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/results-visual/table-header.tsx [hunk 1]
```diff
@@ -159,19 +161,91 @@ export const HeaderCheckbox = ({
...
+
+  const resetColumnWidth = (col: string) => {
+    const columnsWidth = new Map<string, string>([...headerColWidth])
+    columnsWidth.set(col, '200px')
+    setHeaderColWidth(columnsWidth)
+  }
+
+  const removeListeners = React.useCallback(() => {
+    window.removeEventListener('mousemove', mouseMove)
+    window.removeEventListener('mouseup', removeListeners)
```
Comment (shiv-leidos) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/results-visual/table-header.tsx:224]: Please check if it is not a infinite recursive call.

Comment (shiv-leidos) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/results-visual/table-header.tsx:224]: Resolved

**Review by shiv-leidos — APPROVED** [review]

**Review by alexabird — APPROVED** [review]

**Comment by alexabird** [comment]
Retested latest changes, looks good 💯

**Pull request merged**